yellow-paper/docs/decentralization/actors.md

```diff
@@ -0,0 +1,134 @@
+
+1. **Mainstream User**:
+   - Least knowledgeable and least interested in technical details.
+   - Prefers simplicity and ease of use.
+   - Likely to avoid running any node software or only running if bundled
+2. **Enthusiast**: 
+   - More knowledgeable and interested in technology than the mainstream user.
+   - Willing to invest time and resources but not at a professional level.
+3. **Power Users**: 
+   - Technically proficient and deeply engaged.
+   - Likely to have the resources and motivation to run more demanding nodes.
+4. **Developers**: 
+   - Highly knowledgeable with technical skills.
+   - Interested in detailed state and historical data for development and testing.
+5. **Self-Validators**: 
+   - Aim to participate actively in keeping the blockchain available.
+   - Interested in relying on their own nodes and resources (Archive Nodes).
+   - Likely to run nodes that contribute directly to the blockchain's operation, like Sequencers or Provers.
+6. **Institutions**: 
+   - Have significant resources (and potentially technical) capabilities.
+   - Interested in robust participation, potentially running Sequencer or Provers for financial gain.
+
+:::info State-Snapshots
+State snapshots are likely made and published by actors in the **Self-validator** or **Institution** category. These are actors that have the resources to run Archive Nodes and are interested in the historical data.
+:::
+
+### Private eXecution Environment (PXE)
+
+The Aztec PXE is required to do anything meaningful for an end-user. It is the interface between the user and the Aztec node. The PXE is a private enclave that runs on the end-user's machine.
+
+It is responsible for execution contracts and building proofs for the user's transactions privately. This requires the PXE to be connected to a node that can provide the user's state.
+
+:::info Note-discovery
+Note, that as the node will not store secrets for the user. The user relies on some mechanism to figure out "what to lookup", this is what is sometimes referred to as note-discovery: discovering what notes belong to the user.
+:::
+
+As the PXE learns of the users' notes, it stores them in a database for later usage. Beware that the PXE needs to revalidate the membership paths of the notes when they are to be spent. These paths will be obtained from an Aztec node.
+
+
+### Users
+
+From the end-users perspective, they will be interacting with the PXE, which is then talking to the aztec node and the underlying chain.
+
+```mermaid
+graph LR
+   A[User] --> B[PXE]
+   B --> C[Aztec Node]
+   C --> D[Ethereum Node]
+```
+
+The type of node the PXE is connected to depends on the actor type. 
+
+:::info Bundling
+For a mobile application to work nicely for the users the PXE will most likely be bundled with a light node on device. This will allow the user to interact with the system without wider knowledge. Settings might allow you to connect the PXE directly to an endpoint for a node for advanced users.
```

Comment:
I don't think we would bundle a light node on a user's device. At the top you mention that all nodes would connect to the p2p network. We wouldn't have user's devices do this. The PXE on their device would need to call a remote node.

Reply:
Light client would only need to get the header, which for our case could use the P2P similar to what happens for most other blockchains or you could abuse L1 as your p2p layer for getting the header. The light client is there to provide you with stronger guarantees around the header than blind trust.

Bundling it is a way of getting users into using more secure structures. And allows us to have the logic for getting a hold of your notes at the node. In the case that you connect to your own full node, you can simply get it to return values from its local db based on whatever it checked with tags etc. And if light-node the lightnode can call out to services that might exists and validate the returned values against its headers for anchoring. That way the PXE don't need to deal with figuring out how to get the data beyond "just ask the node", and node can then implement what it desires. Similarly to what you can do for things like eth, the PXE or wallet don't care what type of node it is connected to, just that the node provides the expected answers.

Reply:
I think Phil's original point is valid, though, right? I think there are two categories here: a user who only runs a PXE and has to query everything about the blockchain from a 3rd-party server; and a user who runs a PXE & light node on their device. I do think both types of user will exist.

Reply:
@iAmMichaelConnor the desirable end-game of blockchains is that everyone runs a validating light clients and I feel like given that we only need to fetch the headers and that we will have a much lower block frequency then Ethereum then it actually might be feasible to aim for having users use a validating light client (of Aztec) always, no?

Reply:
It would be nice if everyone ran a node. But pragmatically, I don't think that'll happen, so recongnising this class of user seems sensible to me.
If I'm using aztec on my phone, do I want to drain my battery running a light node, or do I want to run the bare minimum software, and make requests to a node only when I need to? I might even be running a full node at home, which I can make requests to. But a resource constrained device doesn't want to stay in sync with the blockchain, I don't think?

Reply:
I suppose I'm not entirely sure what we envisage a 'light' node doing. In my mind, the minimum  amount of functionality for something to be called a 'node' is:

1. Providing world state (hash paths etc).
2. Proving block headers.
3. Providing access to the tx pool's p2p network for transaction submission.

You would never run the above on a mobile device. The p2p chatter and hashing would sap your battery and consume all of your bandwidth.

If what's required is merely a way to independently verify block headers so that I know the node I am connecting to isn't lying to me can that be achieved by configuring the PXE (which is on a user's mobile) to connect to a trusted L1 node and query that information directly?

Reply:
In general light nodes are nodes which have the necessary information to verify that they are not being lied to but don't store any other info. Both on Ethereum and in our case it would be just fetching and hashing the headers. By hashing the headers and hashing the archive tree users get the archive root which they then can verify against the Rollup contract on L1.

Reply:
Updated document to hopefully answer this more ahead of time.